locust/test/test_web.py

```diff
@@ -257,3 +258,37 @@ def my_task(self):
         )
         self.assertEqual(200, response.status_code)
         self.assertIn("Step Load Mode", response.text)
+
+
+class TestWebUIAuth(LocustTestCase):
+    def setUp(self):
+        super(TestWebUIAuth, self).setUp()
+
+        parser = get_parser(default_config_files=[])
+        self.environment.options = parser.parse_args([])
+        self.runner = LocustRunner(self.environment, [])
+        self.stats = self.runner.stats
+
+        self.web_ui = WebUI(self.environment)
+        self.web_ui.app.view_functions["request_stats"].clear_cache()
+        self.setUpBasicAuth()
+        gevent.spawn(lambda: self.web_ui.start("127.0.0.1", 0))
+        gevent.sleep(0.01)
+        self.web_port = self.web_ui.server.server_port
+
+    def setUpBasicAuth(self):
+        self.web_ui.app.config["BASIC_AUTH_ENABLED"] = True
+        self.web_ui.app.config["BASIC_AUTH_USERNAME"] = "john"
+        self.web_ui.app.config["BASIC_AUTH_PASSWORD"] = "doe"
+        auth = BasicAuth()
+        auth.init_app(self.web_ui.app)
+
+    def tearDown(self):
+        super(TestWebUIAuth, self).tearDown()
+        self.web_ui.stop()
+        self.runner.quit()
+
+    def test_index_with_basic_auth_enabled(self):
```

Comment:
This should be two separate tests I think. We should also have a test where incorrect user/pass is supplied.

Reply:
Added multiple tests now (correct, incorrect, blank)